src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -287,7 +287,7 @@ public function beforeRedirect(Event $event, $url, Response $response)
         }
         /** @var \Cake\Controller\Controller $controller */
         $controller = $event->getSubject();
-        $response->body($controller->requestAction($url, [
+        $response->withStringBody($controller->requestAction($url, [
```

Comment:
I dont think this works.
You are not assigning the immutable response.

Reply:
You're sure. Fixed